From 1960e96396f63397c2e32edb7a21f26a1d42ef2e Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 14 Dec 2019 18:39:44 +0200
Subject: [PATCH] [SW-6379] goya2: remove dummy lf2flt symbol

We explicitly use __start as entry point to override GCC wrapper
which looks for lf2flt

Change-Id: I664f563d1940c31618da8f61526cb2daaf97045b
---

diff --git a/Makefile b/Makefile
index 7dcac13..e47db3d 100644
--- a/Makefile
+++ b/Makefile
@@ -23,6 +23,7 @@
 LDFLAGS += -L $(LIBDIR) -lsb -lellipsys
 LDFLAGS += -T $@.lds
 LDFLAGS += -nostartfiles
+LDFLAGS += -Wl,-e__start
 
 # This combination allows removing unused symbols from libsb.a and libellipsys.a
 LDFLAGS += -Wl,--gc-sections
diff --git a/crt0.S b/crt0.S
index ae22f47..34f5c36 100644
--- a/crt0.S
+++ b/crt0.S
@@ -16,10 +16,6 @@
 
 .section .text.__startup, "ax", @progbits
 
-;; lf2flt: to avoid compiler (buildroot wrapper) warning during linkage
-.global	lf2flt
-lf2flt:
-
 .global __start
 .type __start, @function
 .align 4
diff --git a/crt0_rom.S b/crt0_rom.S
index 87a3b76..d7bbfc4 100644
--- a/crt0_rom.S
+++ b/crt0_rom.S
@@ -53,10 +53,6 @@
 
 .section .text.__startup, "ax", @progbits
 
-;; lf2flt: to avoid compiler (buildroot wrapper) warning during linkage
-.global	lf2flt
-lf2flt:
-
 .global __start
 .type __start, @function
 .align 4
